docs/reference/runtime/Execution.md

```diff
@@ -1,6 +1,25 @@
 ## Execution
 
-You can use the same RTOS API in ISR. The only two warnings are:
+### Threads
+
+Your application (`main` function) will start execution in the main thread, but it's not the only thread running in Mbed OS. There's a number of live system threads at any given time. They should be transparent from the user's point of view for most use cases. All threads that exist in the system by default:
+* main - default thread that executes application's `main` function.
+* idle - thread that's run by the scheduler when there's no other activity in the system (e.g. all other threads are waiting for some event). It's used to make sure the board is not burning empty processor cycles, but is put to sleep for as long as possible.
+* timer - thread for handling system and user timer objects.
```

Comment:
Is this really still used? Can we say its depreciated with the RtosTimer class?

Reply:
Good point will add that.